interpreters/luajit/Makefile

```
@@ -0,0 +1,172 @@
+
+include $(APPDIR)/Make.defs
+
+LUAJIT_PATCHS ?= $(sort $(wildcard 000*.patch))
+
+LUAJIT_VERSION := $(patsubst "%",%,$(CONFIG_LUAJIT_VERSION))
+LUAJIT_UNPACK   = LuaJIT
+LUAJIT_TARBALL  = $(LUAJIT_VERSION).tar.gz
+LUAJIT_URL_BASE = https://github.com/LuaJIT/LuaJIT/archive
+LUAJIT_URL      = $(LUAJIT_URL_BASE)/$(LUAJIT_TARBALL)
+
+MAINSRC = luajit.c
+
+CSRCS += lib_aux.c lib_base.c lib_bit.c lib_buffer.c lib_debug.c
+CSRCS += lib_ffi.c lib_init.c lib_io.c lib_jit.c lib_math.c
+CSRCS += lib_os.c lib_package.c lib_string.c lib_table.c
+CSRCS += lj_alloc.c lj_api.c lj_asm.c lj_assert.c lj_bc.c
+CSRCS += lj_bcread.c lj_bcwrite.c lj_buf.c lj_carith.c lj_ccall.c
+CSRCS += lj_ccallback.c lj_cconv.c lj_cdata.c lj_char.c lj_clib.c
+CSRCS += lj_cparse.c lj_crecord.c lj_ctype.c lj_debug.c lj_dispatch.c
+CSRCS += lj_err.c lj_ffrecord.c lj_func.c lj_gc.c lj_gdbjit.c
+CSRCS += lj_ir.c lj_lex.c lj_lib.c lj_load.c lj_mcode.c lj_meta.c
+CSRCS += lj_obj.c lj_opt_dce.c lj_opt_fold.c lj_opt_loop.c
+CSRCS += lj_opt_mem.c lj_opt_narrow.c lj_opt_sink.c lj_opt_split.c
+CSRCS += lj_parse.c lj_prng.c lj_profile.c lj_record.c lj_serialize.c
+CSRCS += lj_snap.c lj_state.c lj_str.c lj_strfmt.c lj_strfmt_num.c
+CSRCS += lj_strscan.c lj_tab.c lj_trace.c lj_udata.c lj_vmevent.c
+CSRCS += lj_vmmath.c
+
+ASRCS += lj_vm.S
+
+VPATH += $(LUAJIT_UNPACK)/dynasm
+VPATH += $(LUAJIT_UNPACK)/src
+VPATH += $(LUAJIT_UNPACK)/src/host
+CFLAGS += -fomit-frame-pointer -fno-short-enums -DLUAJIT_OS=LUAJIT_OS_NUTTX
+CFLAGS += -D__ARM_ARCH_7M__ -DLUAJIT_NO_UNWIND -DLUAJIT_DISABLE_PROFILE 
+CFLAGS += -DLUAJIT_SECURITY_PRNG=0 -DLUAJIT_SECURITY_MCODE=0
+CFLAGS += -DLUA_ROOT=\"/home\"
+
+PROGNAME  = luajit
+PRIORITY  = $(CONFIG_LUAJIT_PRIORITY)
+STACKSIZE = $(CONFIG_LUAJIT_STACKSIZE)
+MODULE    = $(CONFIG_INTERPRETERS_LUAJIT)
+
+$(LUAJIT_TARBALL):
+	$(Q) echo "Downloading $(LUAJIT_TARBALL)"
+	$(Q) curl -O -L $(LUAJIT_URL)
+
+$(LUAJIT_UNPACK): $(LUAJIT_TARBALL)
+	$(Q) echo "Unpacking $(LUAJIT_TARBALL) to $(LUAJIT_UNPACK)"
+	$(Q) tar xzvf $(LUAJIT_TARBALL)
+	$(Q) mv LuaJIT-$(LUAJIT_VERSION) $(LUAJIT_UNPACK)
+	$(Q) cat $(LUAJIT_PATCHS) | \
+		patch -s -N -d $(LUAJIT_UNPACK) -p1
+
+$(LUAJIT_UNPACK)/.patch: $(LUAJIT_UNPACK)
+	$(Q) touch $(LUAJIT_UNPACK)/.patch
+
+DASM_DASC= $(LUAJIT_UNPACK)/src/vm_armv7m.dasc
+
+DASM_FLAGS=
+DASM_FLAGS+= -D ENDIAN_LE
+DASM_FLAGS+= -D JIT
+DASM_FLAGS+= -D FFI
+DASM_FLAGS+= -D DUALNUM
+DASM_FLAGS+= -D FPU
+DASM_FLAGS+= -D HFABI
+DASM_FLAGS+= -D NO_UNWIND
+DASM_FLAGS+= -D VER=70
+DASM_ARCH= armv7m
+
+MINILUA_O= $(LUAJIT_UNPACK)/src/host/minilua.o
+
+BUILDVM_O= $(LUAJIT_UNPACK)/src/host/buildvm.o $(LUAJIT_UNPACK)/src/host/buildvm_asm.o \
+					 $(LUAJIT_UNPACK)/src/host/buildvm_peobj.o $(LUAJIT_UNPACK)/src/host/buildvm_lib.o \
+					 $(LUAJIT_UNPACK)/src/host/buildvm_fold.o
+
+HOSTCFLAGS += -DLUAJIT_DISABLE_PROFILE -I$(LUAJIT_UNPACK)/src -I$(LUAJIT_UNPACK)/src/host -D__ARM_ARCH_7M__=1 -DLUAJIT_TARGET=LUAJIT_ARCH_arm -DLJ_ARCH_HASFPU=1 -DLJ_ABI_SOFTFP=0 -DLUAJIT_NO_UNWIND
```

can we remove -D__ARM_ARCH_7M__=1?

---

let's sort the argument
```suggestion
HOSTCFLAGS += -DLJ_ABI_SOFTFP=0 -DLJ_ARCH_HASFPU=1 -DLUAJIT_DISABLE_PROFILE -DLUAJIT_NO_UNWIND -DLUAJIT_TARGET=LUAJIT_ARCH_arm -I$(LUAJIT_UNPACK)/src -I$(LUAJIT_UNPACK)/src/host
```

---

-D__ARM_ARCH_7M__=1 is neccessary to tell the dynamic assembler how to work with ARMv7-M architecture.